core/templates/pages/exploration-editor-page/modal-templates/state-version-history-modal.component.ts

```diff
@@ -0,0 +1,253 @@
+  getLastEditedVersionNumber(): number {
+    const lastEditedVersionNumber =
+      this.versionHistoryService.getBackwardStateDiffData().oldVersionNumber;
+    if (lastEditedVersionNumber === null) {
+      throw new Error('Last edited version number cannot be null');
+    }
+    return lastEditedVersionNumber;
+  }
+
+  getLastEditedCommitterUsername(): string {
+    return (
+      this.versionHistoryService.getBackwardStateDiffData().committerUsername);
+  }
+
+  // Returns the next version number at which the state was modified.
+  // The explanation for return type being (number | null) is same as explained
+  // above the function getLastEditedVersionNumber().
+  getNextEditedVersionNumber(): number {
+    const nextEditedVersionNumber =
+      this.versionHistoryService.getForwardStateDiffData().oldVersionNumber;
+    if (nextEditedVersionNumber === null) {
+      throw new Error('Next edited version number cannot be null');
+    }
+    return nextEditedVersionNumber;
+  }
+
+  getNextEditedCommitterUsername(): string {
+    return (
+      this.versionHistoryService.getForwardStateDiffData().committerUsername);
+  }
+
+  onClickExploreForwardVersionHistory(): void {
+    this.yamlStrs.previousVersionStateYaml = '';
+    this.yamlStrs.currentVersionStateYaml = '';
+
+    const diffData = this.versionHistoryService.getForwardStateDiffData();
+
+    // The explanation for these if-conditions is added in the below function.
+    if (diffData.newState) {
+      this.newState = diffData.newState;
+      if (diffData.newState.name === null) {
+        throw new Error('State name cannot be null');
+      }
+      this.newStateName = diffData.newState.name;
+    }
+    if (diffData.oldState) {
+      this.oldState = diffData.oldState;
+      if (diffData.oldState.name === null) {
+        throw new Error('State name cannot be null');
+      }
+      this.oldStateName = diffData.oldState.name;
+    }
+    this.committerUsername = diffData.committerUsername;
+    this.oldVersion = diffData.oldVersionNumber;
+
+    this.updateLeftPane();
+    this.updateRightPane();
+
+    this
+      .versionHistoryService
+      .decrementCurrentPositionInStateVersionHistoryList();
+  }
+
+  onClickExploreBackwardVersionHistory(): void {
+    this.yamlStrs.previousVersionStateYaml = '';
+    this.yamlStrs.currentVersionStateYaml = '';
+
+    const diffData = this.versionHistoryService.getBackwardStateDiffData();
+
+    // Explanation for why diffData.newState can be null:
+    // It is explained in VersionHistoryService as to why the values of
+    // newState or oldState can be null. This is because they are elements of
+    // the list fetchedStateData whose last element can be null which marks
+    // the end of the version history of that particular state.
+    if (diffData.newState) {
+      this.newState = diffData.newState;
+      if (diffData.newState.name === null) {
+        throw new Error('State name cannot be null');
```

Comment:
Add explanation above this, similar to what was mentioned in previous comments. Ditto below and elsewhere you have similar checks, so that, if this ever errors in prod, a developer can track down the root cause more easily.

Reply:
Done.